app/facades/foods_facade.rb

```diff
@@ -0,0 +1,19 @@
+class FoodsFacade
+  def self.search_food(query)
+    foods = []
+    data = FoodsService.search_food(query)
+    data[:foods].first(10).each do |food|
+      foods << Food.new(food)
+    end
+  end
+
+  def self.search_food_query(query)
+    data = FoodsService.search_food(query)
+    data[:foodSearchCriteria][:query]
+
+  def self.search_food_total(query)
+    data = FoodsService.search_food(query)
+    data[:totalHits]
+    end
```

Comment:
This would be a great place to use memoization, because you only need to hit the API once with one call to the service. Then, if you had isolated that method, you could use memoization to either hit the API if it was the first time the method runs, or simply return the populated hash if the method had already run once. I'm thinking something like: 
```ruby
def fetch_food_data
data ||= FoodsService.search_food(query)
end
```